include/rx/ranges.hpp

```diff
@@ -1187,142 +1244,66 @@ template <class... Inputs>
-
-/*!
-    @brief Produce sequential groups of exactly N elements, where N is known at compile-time.
-
-    The output type is `std::array<T, N>` rvalue reference, where T is the output type of the
-    inner range (without cvref-qualifiers).
-
-    If the input produces a number of inputs that is not divisible by N, those
-    elements will be skipped.
-*/
-template <size_t N>
-struct in_groups_of_exactly<N> {
-    static_assert(N != 0);
-
-    template <class R>
-    struct Range {
-        R inner;
-        using element_type = remove_cvref_t<typename R::output_type>;
-        using output_type = std::array<element_type, N>&&;
-        static constexpr bool is_finite = is_finite_v<R>;
-        static constexpr bool is_idempotent = true; // we have internal storage
-
-        constexpr explicit Range(R inner) : inner(std::move(inner)) {
-            fill_group();
-        }
-
-        mutable std::array<element_type, N> storage;
-
-        [[nodiscard]] constexpr output_type get() const noexcept {
-            return std::move(storage);
-        }
-
-        [[nodiscard]] constexpr bool at_end() const noexcept {
-            return inner.at_end();
-        }
-
-        constexpr void next() noexcept {
-            inner.next();
-            fill_group();
-        }
-
-        [[nodiscard]] constexpr size_t size_hint() const noexcept {
-            return inner.size_hint() / N;
-        }
-
-        constexpr void fill_group() {
-            // Fill from the current position, but don't increment into the next group.
-            size_t i = 0;
-            while (true) {
-                if (RX_LIKELY(!inner.at_end())) {
-                    storage[i++] = inner.get();
-                    if (RX_LIKELY(i != N)) {
-                        inner.next();
-                    } else {
-                        return;
-                    }
-                } else {
-                    return;
-                }
-            }
-        }
-    };
-
-    template <class R>
-    [[nodiscard]] constexpr auto operator()(R&& input) const {
-        using Inner = get_range_type_t<R>;
-        return Range<Inner>{as_input_range(std::forward<R>(input))};
-    }
-};
-
 /*!
-    @brief Produce sequential groups of exactly N elements, where N is only known at runtime.
+    @brief Produce sequential groups of exactly N elements.
 
-    The output type is `std::vector<N>` const reference, where T is the output type of the inner
-    range (without cvref-qualifiers), and the size of the result is always N.
+    The output type is a range of exactly N element.
 
-    If the input produces a number of inputs that is not divisible by N, those elements will be
+    If the input produces a number of inputs that is not divisible by N, elements at the end will be
     skipped.
 */
-template <>
-struct in_groups_of_exactly<> {
-    size_t n;
+struct in_groups_of_exactly {
+    const size_t n;
 
     constexpr explicit in_groups_of_exactly(size_t n) noexcept : n(n) {}
```

Comment:
`n == 0` is possible ? `in_groups_of ` throw an exception.

Reply:
Yep, that was removed by mistake.